cli/create_fuzzer_instance_from_argv.cpp

```diff
@@ -25,40 +25,48 @@
 #include "fuzzuf/logger/log_file_logger.hpp"
 #include "fuzzuf/logger/logger.hpp"
 #include "fuzzuf/logger/stdout_logger.hpp"
+#include "fuzzuf/utils/check_crash_handling.hpp"
 
 namespace fuzzuf::cli {
 
-std::unique_ptr<Fuzzer> CreateFuzzerInstanceFromArgv(int argc, const char **argv) {
-    // Explicitly enable logging to stdout as Logger does not get confirmed before parsing command line options
-    StdoutLogger::Enable();
+std::unique_ptr<Fuzzer> CreateFuzzerInstanceFromArgv(int argc,
+                                                     const char **argv) {
+  // Explicitly enable logging to stdout as Logger does not get confirmed before
+  // parsing command line options
+  StdoutLogger::Enable();
 
-    GlobalFuzzerOptions global_options;
+  GlobalFuzzerOptions global_options;
 
-    GlobalArgs global_args = {.argc = argc, .argv = argv};
-    FuzzerArgs fuzzer_args =
-        ParseGlobalOptionsForFuzzer(global_args, /* &mut */ global_options);
+  GlobalArgs global_args = {.argc = argc, .argv = argv};
+  FuzzerArgs fuzzer_args =
+      ParseGlobalOptionsForFuzzer(global_args, /* &mut */ global_options);
 
-    // Follow the command line, and initialize a logger instance which gets and saves the logs
-    StdoutLogger::Disable();
-    if (global_options.logger == Logger::Stdout) {
-        StdoutLogger::Enable();
-    } else if (global_options.logger == Logger::LogFile) {
-        if (global_options.log_file.has_value()) {
-            DEBUG("LogFile logger is enabled");
-            LogFileLogger::Init(global_options.log_file.value());
-        } else {
-            throw exceptions::cli_error("LogFile logger is specified, but log_file "
-                                        "is not specified. May be logic bug",
-                                        __FILE__, __LINE__);
-        }
+  // Follow the command line, and initialize a logger instance which gets and
+  // saves the logs
+  StdoutLogger::Disable();
+  if (global_options.logger == Logger::Stdout) {
+    StdoutLogger::Enable();
+  } else if (global_options.logger == Logger::LogFile) {
+    if (global_options.log_file.has_value()) {
+      DEBUG("LogFile logger is enabled");
+      LogFileLogger::Init(global_options.log_file.value());
     } else {
-        throw exceptions::cli_error("Unsupported logger: " +
-                                      to_string(global_options.logger),
-                                    __FILE__, __LINE__);
+      throw exceptions::cli_error(
+          "LogFile logger is specified, but log_file "
+          "is not specified. May be logic bug",
+          __FILE__, __LINE__);
     }
+  } else {
+    throw exceptions::cli_error(
+        "Unsupported logger: " + to_string(global_options.logger), __FILE__,
+        __LINE__);
+  }
+
+  utils::CheckCrashHandling();
```

Comment:
I think that, ideally, `CheckCrashHandling` should be called in the constructors of Executors or other places than here since crash handling mechanism may be absent in special execution environments like emulators. But I think there is no problem now because we don't have such executors and because this would be at most a minor issue.

Nevertheless, maybe we should leave comments here like "NOTE: move this call to somewhere else if we implment a new executor in which crash handling doesn't exist."

Reply:
Since qemu executor is such a special execution environment, the check is moved to constructor of affected executors.

Reply:
Is it? 
I think user mode QEMU emits SIGSEGV when an emulated binary emits SIGSEGV like this:

```
~/qemu/qemu-5.0.0/build/x86_64-linux-user$ cat > test.c
#include <stdio.h>

int main() {
  char *p = 0x0;
  *p = 0;
  puts(p);    
}
~/qemu/qemu-5.0.0/build/x86_64-linux-user$ gcc test.c -o test
~/qemu/qemu-5.0.0/build/x86_64-linux-user$ ./qemu-x86_64 ./test
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
Segmentation fault (core dumped)
```

Actually I don't know how QEMUExecutor is implemented and used in fuzzuf, so I would like you to double-check it considering this.

Reply:
It seems you are right. The check is added to the constructor of QEMUExecutor.